tools/mason/MasonPublish.chpl

```diff
@@ -250,24 +307,37 @@ proc getPackageName() throws {
 /* Adds package to the Bricks of the mason-registry branch and then adds the version.toml
  with the source url of the package's GitHub repo.
  */
-private proc addPackageToBricks(projectLocal: string, safeDir: string, name : string) throws {
+private proc addPackageToBricks(projectLocal: string, safeDir: string, name : string,registryPath : string, isLocal : bool) throws {
   try! {
-
     const toParse = open(projectLocal+ "/Mason.toml", iomode.r);
-    const url = gitUrl();
     var tomlFile = new owned(parseToml(toParse));
     const versionNum = tomlFile['brick']['version'].s;
-    mkdir(safeDir + "/mason-registry/Bricks/" + name);
-    const baseToml = tomlFile;
-    var newToml = open(safeDir + "/mason-registry/Bricks/" + name + "/" + versionNum + ".toml", iomode.cw);
-    var tomlWriter = newToml.writer();
-    baseToml["brick"].set("source", url[1..url.length-1]);
-    tomlWriter.write(baseToml);
-    tomlWriter.close();
+
+    if !isLocal {
+      mkdir(safeDir + "/mason-registry/Bricks/" + name);
+      const baseToml = tomlFile;
+      var newToml = open(safeDir + "/mason-registry/Bricks/" + name + "/" + versionNum + ".toml", iomode.cw);
+      var tomlWriter = newToml.writer();
+      const url = gitUrl();
+      baseToml["brick"].set("source", url[1..url.length-1]);
+      tomlWriter.write(baseToml);
+      tomlWriter.close();
+    }
+    else {
+      mkdir(safeDir + "/Bricks/" + name);
+      const baseToml = tomlFile;
+      var newToml = open(safeDir + "/Bricks/" + name + "/" + versionNum + ".toml", iomode.cw);
+      var tomlWriter = newToml.writer();
+      baseToml["brick"].set("source", projectLocal);
+      tomlWriter.write(baseToml);
+      tomlWriter.close();
+      const gitMessageString = ('git tag -a v' + versionNum + ' -m  "' + name + '"');
+      gitC(projectLocal, gitMessageString);
     }
+  }
   catch {
     writeln('ERROR: ' + name + ' already exists in the Bricks');
```

Comment:
3. I get this error everytime I try to publish a brand new package

```bash
 [spartee::~]
$ rm -rf .mason
 [spartee::~]
$ cd chapel/external-pkgs/
 [spartee:mason-publish-:~/chapel/external-pkgs]
$ mason new newnewnew
Created new library project: newnewnew
 [spartee:mason-publish-:~/chapel/external-pkgs]
$ cd newnewnew/
 [spartee:master:~/chapel/external-pkgs/newnewnew]
$ mason build
Updating mason-registry
Compiling [debug] target: newnewnew
Build Successful

 [spartee:master:~/chapel/external-pkgs/newnewnew]
$ mason publish ~/.mason
ERROR: newnewnew already exists in the Bricks
```

Reply:
This is mainly an issue of that error message not being updated, but this use case should fail because ~/.mason is not a proper local registry, Ill update the error message but I am fairly certain this should always fail

Reply:
Ahh you are correct, I will adapt my review.